src/CodeGen_Hexagon.cpp

```diff
@@ -1355,6 +1355,33 @@ void CodeGen_Hexagon::visit(const Call *op) {
             Value *idx = codegen(op->args[1]);
             value = vlut(lut, idx, *min_index, *max_index);
             return;
+        } else if (op->is_intrinsic(Call::select_mask)) {
+            internal_assert(op->args.size() == 3);
+            // eliminate_bool_vectors has replaced all boolean vectors
+            // with integer vectors of the appropriate size, so we
+            // just need to convert the select_mask intrinsic to a
+            // hexagon mux intrinsic.
+            value = call_intrin(op->type,
+                                "halide.hexagon.mux" +
+                                type_suffix(op->args[1], op->args[2], false),
+                                op->args);
+            return;
+        } else if (op->is_intrinsic(Call::bool_to_mask)) {
+            internal_assert(op->args.size() == 1);
+            if (op->args[0].type().is_vector()) {
+                // The argument is already a mask of the right width. Just
+                // sign-extend to the expected type.
+                op->args[0].accept(this);
+            } else {
+                // The argument is a scalar bool. Converting it to
+                // all-ones or all-zeros is sufficient for HVX masks
+                // (mux just looks at the LSB).
```

Comment:
To be precise, it looks at the LSB of each byte, (otherwise negating wouldn't be necessary).